back/node_watcher/src/nodeWatcher.ts

```diff
@@ -81,11 +81,14 @@ async function incrementor(
       api.registry.setMetadata(rpcMeta);
     }
 
+    const events = await api.query.system.events.at(blockHash);
+    const sessionIndex = await api.query.session.currentIndex.at(blockHash);
+
     // execute watcher tasks
     for await (const task of tasks) {
       l.warn(`Task --- ${task.name}`);
 
-      const result = await task.read(blockHash, api);
+      const result = await task.read(blockHash, events, sessionIndex, api);
```

Comment:
Could also consider to put these 2 args in a `cached: {events, sessionIndex}` object, so that you don't end up with 7 arguments soon

Reply:
will do